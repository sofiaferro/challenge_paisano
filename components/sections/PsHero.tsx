import React from 'react';

import styled from 'styled-components';

import { theme } from '@/styles/theme';
import texts from '@/styles/texts';

import PsButton from '../molecules/PsButton';

interface PsHeroProps {
  action: () => void;
}

// styles
const { h1, p } = texts;
const { firstBg } = theme;

const Container = styled.div`
  ${firstBg};
  width: 100%;
  height: 40vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const Title = styled.h1`
  ${h1};
  padding: 0.5em;
`;

const Text = styled.p`
  ${p};
  text-transform: uppercase;
  font-size: 12px;
  @media (max-width: 680px) {
    font-size: 10px;
  }
`;

const PsHero = ({ action }: PsHeroProps) => {
  // data
  const data = {
    title: 'The new creative economy.',
    text: 'Create, explore, & SELL digital art NFTs.',
  };
  return (
    <Container>
      <Text>{data.title}</Text>
      <Title>{data.text}</Title>
      <PsButton title={'Explore'} onClick={action} />
    </Container>
  );
};

export default PsHero;
