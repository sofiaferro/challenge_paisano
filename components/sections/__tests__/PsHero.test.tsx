import React from 'react';
import renderer from 'react-test-renderer';

import { describe, expect, it } from '@jest/globals';

import PsHero from '../PsHero';
describe('PsHero', () => {
  it('renders the component', () => {
    const mockCallback = jest.fn();
    const tree = renderer.create(<PsHero action={mockCallback} />);
    expect(tree).toMatchSnapshot();
  });
});
