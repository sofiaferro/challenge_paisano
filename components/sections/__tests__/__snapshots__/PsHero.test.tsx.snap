// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PsHero renders the component 1`] = `
<div
  className="sc-25d04c34-0 ipzWyM"
>
  <div
    className="sc-25d04c34-1 QXRnW"
  >
    <p
      className="sc-25d04c34-3 koPsWP"
    >
      Create, explore, & SELL digital art NFTs.
    </p>
    <h3
      className="sc-25d04c34-2 iqLTwI"
    >
      The new creative economy.
    </h3>
    <div
      className="sc-21b4ebd5-1 gPuGRa"
      style={{}}
    >
      <div
        className="sc-21b4ebd5-0 lmlyrg"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={{}}
      >
        <p>
          Explore
        </p>
      </div>
    </div>
  </div>
</div>
`;
